frontend/app/forms/motion-to-reduce-conviction/motion-to-reduce-conviction.pdf.component.js

```diff
@@ -1,5 +1,127 @@
+        <PositionedString debugKey="person.fullName" left="10.5%" top="14.1%">
+          <div>
+            {renderData("person.firstName")} {renderData("person.middleName")}{" "}
+            {renderData("person.lastName")}
+          </div>
+        </PositionedString>
+        <PositionedString
+          dataKey="person.addressStreet"
+          left="11.65%"
+          top="17.49%"
+        />
+        <PositionedString
+          debugKey="petitionerAddress"
+          left="11.65%"
+          top="21.09%"
+        >
+          {data.person.addressCity && `${renderData("person.addressCity")}, `}
+          {renderData("person.addressState")}
+          {`\u0020`}
+          {renderData("person.addressZip")}
+        </PositionedString>
+        <PositionedString
+          dataKey="person.homePhone"
+          left="11.65%"
+          top="24.5%"
+        />
+        <PositionedString dataKey="person.email" left="11.65%" top="28.1%" />
+        <PositionedCheckmark
+          dataKey="person.petitionerRepresentative"
+          left="20%"
+          top="31.5%"
+          shouldShow={data.person.petitionerRepresentative === "petitioner"}
+        />
+        <PositionedCheckmark
+          dataKey="person.petitionerRepresentative"
+          left="20%"
+          top="33%"
+          shouldShow={data.person.petitionerRepresentative === "attorney"}
+        />
+        <PositionedString
+          dataKey="person.petitionerBarNumber"
+          left="62%"
+          top="33%"
+        />
+        <PositionedCheckmark
+          dataKey="case.courtType"
+          left="36%"
+          top="36.9%"
+          shouldShow={data.case.courtType === "District"}
+        />
+        <PositionedCheckmark
+          dataKey="case.courtType"
+          left="46.9%"
+          top="36.9%"
+          shouldShow={data.case.courtType === "Justice"}
+        />
+        <PositionedString dataKey="case.courtAddress" left="27.8%" top="43%" />
+        <PositionedString left="25.9%" top="40.2%" debugKey="judicialDistrict">
+          {data.case.courtAddress &&
+            getJudicialDistrict(data.case.courtAddress, data.case.courtType)}
+        </PositionedString>
+        <PositionedString left="50.7%" top="40.2%" debugKey="county">
+          {data.case.courtAddress &&
+            getCounty(data.case.courtAddress, data.case.courtType)}
+        </PositionedString>
+        <PositionedString dataKey="case.caseNumber" left="54.1%" top="57.9%" />
+        <PositionedString dataKey="case.judgeName" left="54.1%" top="62.5%" />
+        <PositionedString dataKey="case.plaintiff" left="11%" top="50.7%" />
+        <PositionedString dataKey="case.defendant" left="11%" top="58.1%" />
+        {requestedConvictionDegreeInformation(
+          data.case.convictionDegree,
+          data.case.reducedConvictionDegree
+        )}
+      </RenderPage>
+    </>
+  );
+}
+
+function requestedConvictionDegreeInformation(degA, degB) {
```

Comment:
this render function handles the (1) section [here](https://www.utcourts.gov/howto/criminallaw/docs/01_Motion_to_Reduce_Conviction.pdf). The positioning of the values should change depending on whether the petitioning for one or two degree reduction. I'd love feedback on this logic and whether there can be improvements.

Reply:
I like how you wrote this function. I think it's pretty easy to read and follow. One question is whether a "Fourth Degree" ever exists for any felonies/misdemeanors? I have never heard of that, but if that exists then our logic will think the fourth degree is actually a third degree. Might be worth checking with @tuckersamuelsen on this one.

Reply:
Yes, it would be nice to get @tuckersamuelsen input on this. I found this [doc](https://www.utcourts.gov/howto/criminallaw/penalties.asp) that outlines classifications of criminal offenses in Utah. It appears there are these degrees of felonies:
1. Capital
2. First
3. Second
4. Third

Question **(1)** in [motion to reduce conviction](https://www.utcourts.gov/howto/criminallaw/docs/01_Motion_to_Reduce_Conviction.pdf) gives 2 options; your either requesting reduce your conviction by **one** or **two** degrees, there is no option for reduce your conviction by **three** degrees. This made me think that perhaps this from only pertains to people with either first, second, or third degree felonies?

Reply:
Yeah I am unaware of how Capital offenses are treated in the reduction of conviction process. It seems unlikely that it would be an easy process to reduce a capital offense.

How are capital offenses being treated right now by your code?

Reply:
Sorry late to this.  Here's some general info about the 402 process.

402s also apply to misdemeanors and infractions, so there's really 8 classifications: Capital, Felony 1, 2, and 3, Misdemeanor A, B, C, and infraction, in that order.  We'll never have to worry about Capital so we could probably just exclude that.

You can reduce it by a *maximum* of two steps - i.e., Felony 1 to Felony 3, or Felony 3 to Misdemeanor B, or MB to Infraction, etc.  Even then, the prosecutor has to specifically sign off to reduce by two steps - it's tougher and rarer, but it does happen and it never hurts to try.

Misdemeanor reductions are very important for expungement eligibility.  You have a lifetime limit of convictions before you aren't eligible, but if you can reduce a few infractions to misdemeanors than you might be.

We'll probably want to ask the user if they request a hearing, but generally they would probably prefer not to.  In Section B, you'd actually want to check both boxes if they want a 2-step reduction - that tells the court they'd like a 2-step but will settle for a 1-step.

Reply:
Amazing. Thanks @tuckersamuelsen. so exclude capital from this form.

Currently I only have the 3 degrees of felonies as options. 

I will add the additional 4 misdemeanors to the options on the web form. 

I will also work on adding validation logic for this (I think if the petitioner has a 1st degree felony he should only have  2nd, and 3rd degree felonies as options, not the entire 7 classifications of conviction).

Reply:
Yes, that's correct.  It's never possible to reduce by 3 steps but it might confuse the user to have additional options.  So it should only allow them to choose 1- and 2-steps.

Thanks!